diff --git a/package.json b/package.json
index b31ccc660d05d9201e57ae7f82dfbe0af0057441..842b2dd8f00f12f8929e563851f45475d155e109 100644
--- a/package.json
+++ b/package.json
@@ -6,6 +6,7 @@
   "main": "./lib/cjs/index.js",
   "exports": {
     ".": {
+      "types": "./typings.d.ts",
       "require": "./lib/cjs/index.js",
       "default": "./lib/esm/index.mjs"
     }
diff --git a/typings.d.ts b/typings.d.ts
index 580c56f4397ad8d7ba8a9527e95bbe3e973802b8..b52e940f6caec0dd9330e62cd16ef881e1452e00 100644
--- a/typings.d.ts
+++ b/typings.d.ts
@@ -1,6 +1,6 @@
 import type { ErrorObject } from 'ajv';
 
-export interface IOutputError {
+interface IOutputError {
   start: { line: number; column: number; offset: number };
   // Optional for required
   end?: { line: number; column: number; offset: number };
@@ -8,7 +8,7 @@ export interface IOutputError {
   suggestion?: string;
 }
 
-export interface IInputOptions {
+interface IInputOptions {
   format?: 'cli' | 'js';
   indent?: number | null;
 
@@ -16,9 +16,15 @@ export interface IInputOptions {
   json?: string | null;
 }
 
-export default function <S, T, Options extends IInputOptions>(
+declare function betterAjvErrors<S, T, Options extends IInputOptions>(
   schema: S,
   data: T,
   errors: Array<ErrorObject>,
   options?: Options
 ): Options extends { format: 'js' } ? Array<IOutputError> : string;
+
+declare namespace betterAjvErrors {
+  export { IOutputError, IInputOptions };
+}
+
+export = betterAjvErrors;
